cis194/week2/robert-chang/src/LogAnalysis.hs

```diff
@@ -0,0 +1,39 @@
+{-# OPTIONS_GHC -Wall #-}
+
+module LogAnalysis where
+import Data.Maybe (isJust, fromJust)
+import Text.Read ( readMaybe )
+import Log
+
+-- Exercise 1
+
+-- Helpers
+safeIntReader :: String -> Maybe Int
+safeIntReader = readMaybe
+
+getIntFromJust :: String -> Int 
+getIntFromJust s = fromJust . safeIntReader $ s 
+
+isJustValue :: String -> Bool
+isJustValue s = isJust . safeIntReader $ s
+
+parseMessage :: String -> LogMessage
+parseMessage s = case words s of 
+             ("E": level : timestamp : msg)  -> 
+               if (and [isJustValue level, isJustValue timestamp])
+                  then LogMessage (Error $ getIntFromJust $ level) (getIntFromJust timestamp) (unwords msg)
```

Comment:
Nit: align `if`, `then` and `else`
```Haskell
if ....
then ....
else ....
```

Also, when we check if something is `Just ...` then use `fromJust`, we can always use pattern match instead.
You can have nested pattern match. Or an advance trick is to use view patterns extension: https://gitlab.haskell.org/ghc/ghc/wikis/view-patterns

Reply:
This is mind blowing...